lib/strings.nix

```diff
@@ -205,7 +205,12 @@ rec {
        normalizePath "/a//b///c/"
        => "/a/b/c/"
   */
-  normalizePath = s: (builtins.foldl' (x: y: if y == "/" && hasSuffix "/" x then x else x+y) "" (stringToCharacters s));
+  normalizePath = s:
+    if isPath s then throw ''
+      lib.strings.normalizePath: The argument (${toString s}) is not a string but a path, which is not supported.
+          Values of the path data type are always normalised in Nix, there's no need to do that manually.
+          To normalise subpaths when working with path data types, please use `lib.path.subpath.normalise`.''
```

Comment:
What's a _path data type_?

Reply:
A path in the data type listing for Nix: https://nixos.org/manual/nix/stable/language/values.html?type-path#type-path

I find just "Paths are always normalised in Nix" a bit too confusing, given that we can have paths in strings too.

Reply:
Now changed this to "Path values are always normalised in Nix", which is what you also used in your other suggestions, I think that sounds good.

Reply:
Sounds good. I had an inkling but my mind did a 180 at "data type" which is a bit outside the usual lingo.